### ISSUE TITLE
superset behaviour in matrix_norm (linear_algebra.py)

### ISSUE THREAD

**Issue by randomstate42** [unifyai/ivy#5402]
after adding superset behavoir to matrix_norm , the type hint in `axis: Optional[Union[int, Sequence[int]]] = None` gives the possibility to use an integer while the backend implemetations convert the axis to a tuple, in the case of an integer axis that would throw the error "int object is not iterable". Besides, in the original backend implementations, when the axis is an integer the vector norms are calculated instead while the ivy implementation separates the vector and matrix norms implementations, following the python array API standard . So I think there is no need for the int type hint. Also there is a mismatch between default values for the _axis_ argument in [Container, array, functional api] implementations (which is **None**) and the default values in backend implementations (which is **(-2, -1)**). This doesn't throw an error but I don't know if it's the convenient way.

**Comment by randomstate42** [comment]
https://github.com/unifyai/ivy/blob/8db7a9f04a0dcf7cfd26384c646278f12a67a5c8/ivy/functional/ivy/linear_algebra.py#L936

**Comment by randomstate42** [comment]
also only numpy and jax backend implementations require a convertion of the sequence to a tuple. Tensorflow and pytorch handle that in the original implementattions.
